src/kafka-file/codeLensProvider.ts

```diff
@@ -0,0 +1,223 @@
+    }
+
+    private isSeparator(line: string): boolean {
+        return line === "###";
+    }
+
+    public dispose(): void {
+        this.consumerCollection.dispose();
+        this.disposables.forEach(d => d.dispose());
+    }
+
+    private createBlockLens(blockStartLine: number, blockEndLine: number, document: vscode.TextDocument, blockType: BlockType): vscode.CodeLens[] {
+        const range = new vscode.Range(blockStartLine, 0, blockEndLine, 0);
+        const lineRange = new vscode.Range(blockStartLine, 0, blockStartLine, 0);
+        if (blockType === BlockType.CONSUMER) {
+            return this.createConsumerLens(lineRange, range, document);
+        }
+        return this.createProducerLens(lineRange, range, document);
+    }
+
+    private createProducerLens(lineRange: vscode.Range, range: vscode.Range, document: vscode.TextDocument): vscode.CodeLens[] {
+        const produceRecordCommand = this.createProduceRecordCommand(document, range);
+        return [
+            new vscode.CodeLens(lineRange, {
+                title: "Produce record",
+                command: ProduceRecordCommandHandler.COMMAND_ID,
+                arguments: [produceRecordCommand, 1]
+            }),
+            new vscode.CodeLens(lineRange, {
+                title: "Produce record x 10",
+                command: ProduceRecordCommandHandler.COMMAND_ID,
+                arguments: [produceRecordCommand, 10]
+            })
+        ];
+    }
+
+    private createProduceRecordCommand(document: vscode.TextDocument, range: vscode.Range): ProduceRecordCommand {
+        let topicId;
+        let key;
+        let value = "";
+        for (let currentLine = range.start.line + 1; currentLine <= range.end.line; currentLine++) {
+            const line = document.lineAt(currentLine);
+
+            if (line.text.startsWith("topic:")) {
+                topicId = line.text.substr("topic:".length).trim();
+                continue;
+            }
+
+            if (line.text.startsWith("key:")) {
+                key = line.text.substr("key:".length).trim();
+                continue;
+            }
+
+            if (line.text.startsWith("--")) {
+                continue;
+            }
+
+            value = document.getText(new vscode.Range(currentLine, 0, range.end.line + 1, 0)).trim();
+            break;
+        }
+
+        return {
+            topicId,
+            key,
+            value,
+        };
+    }
+
+    private createConsumerLens(lineRange: vscode.Range, range: vscode.Range, document: vscode.TextDocument): vscode.CodeLens[] {
+        const launchCommand = this.createLaunchConsumerCommand(document, range, this.clusterSettings.selected?.id);
+        const consumeUri = createConsumerUri(launchCommand);
+        const started = this.consumerCollection.has(consumeUri);
+        return [
+            new vscode.CodeLens(lineRange, {
+                title: `${started ? "$(check)" : "$(x)"}`,
+                command: ""
+            }),
```

Comment:
if we only show "start  consumer" when it's stopped, and "stop consumer" when it's started, then this code lens is redundant and can be removed

Reply:
I think it's better to have 2 codelens for status and Start/Stop, because when you hover the codens, the status codel lens should not be highlighted